ops/model.py

```diff
@@ -1400,6 +1414,27 @@ def get_pebble(self, socket_path: str) -> 'pebble.Client':
         """Create a pebble.Client instance from given socket path."""
         return pebble.Client(socket_path=socket_path)
 
+    def add_pebble_layer_file(self, layers_dir, layer_yaml, layer_name='layer'):
```

Comment:
I don't particularly like that this can take a layer_name, but has no way to supply it from higher levels.
Also, if we are going down this route, I think we need to expose the various layers to the client.
If there are layers 001, 002, 003, 004. What are you actually adding when you add 005, unless you know what is going on in 1-4?
I think at one point we were using a collapsed view (read all of them, and apply a new one on top).
That feels like we are backing ourselves into a poorly scaling solution. If there ever is a lot of churn between the charm and its associated application (if you need to run actions via pebble requests, for example), then the layers keep building and building.
(having to read 800 layer files to figure out what is going on feels like we're far away from a useful interaction.)